avahi-core/domain-util.c

```diff
@@ -82,6 +85,14 @@ static int load_lsb_distrib_id(char *ret_s, size_t size) {
     fclose(f);
     return -1;
 }
+
+static int load_lsb_distrib_id(char *ret_s, size_t size) {
+    return load_id(ret_s, size, "/etc/lsb-release", "DISTRIB_ID=");
+}
+
+static int load_os_id(char *ret_s, size_t size) {
+    return load_id(ret_s, size, "/etc/os-release", "ID=");
```

Comment:
Oh, there is defined also ``DEFAULT_HOSTNAME=`` variable in man os-release, which Fedora distribution actually uses. It is now the same value as ID= and only optional. But I think we use that identification for hostname, so we should attempt to use that hostname if present. This is defined exactly for our use case. I think we should use DEFAULT_HOSTNAME from that file if present, ID as a fallback. Only ``ID=`` is present on Debian. But then the function should be called *load_os_hostname* or *load_os_default_hostname*.